client/blocks/signup-form/index.jsx

```diff
@@ -697,21 +699,26 @@ class SignupForm extends Component {
 						) }
 					</>
 				) }
-
-				<FormLabel htmlFor="password">{ this.props.translate( 'Choose a password' ) }</FormLabel>
-				<FormPasswordInput
-					className="signup-form__input"
-					disabled={ this.state.submitting || this.props.disabled }
-					id="password"
-					name="password"
-					value={ formState.getFieldValue( this.state.form, 'password' ) }
-					isError={ formState.isFieldInvalid( this.state.form, 'password' ) }
-					isValid={ formState.isFieldValid( this.state.form, 'password' ) }
-					onBlur={ this.handleBlur }
-					onChange={ this.handleChangeEvent }
-					submitting={ this.state.submitting || this.props.submitting }
-				/>
-				{ this.passwordValidationExplanation() }
+				{ this.props.displayPasswordInput && (
+					<>
+						<FormLabel htmlFor="password">
+							{ this.props.translate( 'Choose a password' ) }
+						</FormLabel>
+						<FormPasswordInput
+							className="signup-form__input"
+							disabled={ this.state.submitting || this.props.disabled }
+							id="password"
+							name="password"
+							value={ formState.getFieldValue( this.state.form, 'password' ) }
+							isError={ formState.isFieldInvalid( this.state.form, 'password' ) }
+							isValid={ formState.isFieldValid( this.state.form, 'password' ) }
+							onBlur={ this.handleBlur }
+							onChange={ this.handleChangeEvent }
+							submitting={ this.state.submitting || this.props.submitting }
+						/>
+						{ this.passwordValidationExplanation() }
+					</>
+				) }
```

Comment:
The `SignupForm` is a component used in various parts that will not have the new `displayPasswordInput` property. In that case, the password field would disappear.

`SignupForm` already uses `PasswordlessSignupForm` if `isPasswordless` is true, we should try to leverage that.

Reply:
I have pushed a version handling it. Please, let me know what you think about it.